tests/test_lion8b.py

```diff
@@ -0,0 +1,422 @@
+    torch.manual_seed(123)
+    W = torch.rand((D, D), device=device, requires_grad=True)
+    with torch.no_grad():
+        W += torch.sign(W)  # bound away from zero so decay won't change sign
+    W_orig = W.detach().clone()
+
+    opt = Lion8bit([W],
+                   _fused=fused,
+                   betas=(.5, .5),
+                   quantize=(device != 'cpu'),
+                   weight_decay=weight_decay,
+                   error_correction=use_errors)
+
+    zeros_grad = torch.zeros_like(W)
+    for _ in range(5):
+        W.grad = zeros_grad
+        opt.step()
+        opt.zero_grad()
+
+        mom = opt.state[W]['exp_avg']  # type:ignore using tensor as key
+        assert torch.all(mom.materialize() == 0)
+        if mom.is_quantized():
+            assert torch.all(mom.quantized == 0)
+
+        if weight_decay:
+            assert torch.all(W_orig.abs() > W.abs())
+        else:
+            torch.testing.assert_close(W_orig, W)  # no weight modification
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize('N,D', [(1, 8), (17, 23), (32, 32)])
+@pytest.mark.parametrize('device', ['cpu', 'cuda'])
+@pytest.mark.parametrize('dtype', _FLOAT_DTYPES)
+@pytest.mark.parametrize('fused,use_errors', [(False, False), (True, False),
+                                              (True, True)])
+def test_descends(N: int, D: int, device: str, dtype: torch.dtype, fused: bool,
+                  use_errors: bool) -> None:
+    if (dtype != torch.float32) and device == 'cpu':
+        return
+    torch.manual_seed(123)
+    X = torch.randn((N, D), device=device, requires_grad=False, dtype=dtype)
+    W = torch.randn((D, D), device=device, requires_grad=True, dtype=dtype)
+
+    # we use tiny beta1 so we move almost entirely in the gradient direction
+    opt = Lion8bit([W],
+                   lr=1e-2,
+                   betas=(.5, .5),
+                   quantize=(device != 'cpu'),
+                   _fused=fused,
+                   error_correction=use_errors)
+
+    prev_loss = np.inf
+    prev_momentum = None
+    num_iters = 10 if device == 'cuda' else 2  # keep test fast
+    for _ in range(num_iters):
+        Y = X @ W
+        loss = (Y * Y).mean()
+        loss.backward()
+        opt.step()
+        opt.zero_grad()
+
+        loss_val = loss.item()
+        assert loss_val < prev_loss
+        prev_loss = loss_val
+
+        # since we're getting the same batch every time and have a small
+        # learning rate, our gradients should point in the same direction
+        # at each step. Consequently, our momentum should grow each step.
+        state_for_param = opt.state[W]  # type:ignore using tensor as key
+        momentum = state_for_param['exp_avg'].materialize()
+        assert momentum is not None and momentum.shape == W.shape
+        if prev_momentum is not None:
+            momentum_changes = momentum - prev_momentum
+            assert torch.all(momentum_changes >= 0)
+            assert momentum_changes.max() > 0
```

Comment:
```
        # since we're getting the same batch every time and have a small
        # learning rate, our gradients should point in the same direction
        # at each step. Consequently, our momentum should grow each step.
```

that describes why its norm or abs should grow, not why it should always be positive.